sanic/server/websockets/impl.py

```diff
@@ -840,3 +844,10 @@ def connection_lost(self, exc):
         self.abort_pings()
         if self.connection_lost_waiter:
             self.connection_lost_waiter.set_result(None)
+
+    async def __aiter__(self):
+        while True:
+            try:
+                yield await self.recv()
+            except ConnectionClosedOK:
+                return
```

Comment:
I didn't benchmark that but I believe this can be faster if the try-except is moved outside of the loop. Someone else needs to look at whether this is the handling that we want for closed sockets.

Reply:
For cleaner code avoiding try-except for this sort of structure, consider:
```python
from contextlib import suppress

with suppress(ConnectionClosedOK, AnyOtherExceptionToIgnore, ...):
    while True:
        yield await self.recv()
```

Reply:
Thank you! My first time to use `contextlib`. It looks very useful.